components/admin_console/billing/delete_workspace/delete_workspace_modal.tsx

```diff
@@ -0,0 +1,354 @@
+import DeleteFeedbackModal from 'components/admin_console/billing/delete_workspace/delete_feedback';
+import DowngradeFeedbackModal from 'components/admin_console/billing/delete_workspace/downgrade_feedback';
+import {Feedback} from '@mattermost/types/cloud';
+import {GlobalState} from 'types/store';
+import useGetUsage from 'components/common/hooks/useGetUsage';
+import {fileSizeToString} from 'utils/utils';
+import useOpenDowngradeModal from 'components/common/hooks/useOpenDowngradeModal';
+import {subscribeCloudSubscription, deleteWorkspace as deleteWorkspaceRequest} from 'actions/cloud';
+import ErrorModal from 'components/cloud_subscribe_result_modal/error';
+import DeleteWorkspaceProgressModal from 'components/admin_console/billing/delete_workspace/progress_modal';
+import SuccessModal from 'components/cloud_subscribe_result_modal/success';
+import ResultModal from 'components/admin_console/billing/delete_workspace/result_modal';
+import PaymentSuccessStandardSvg from 'components/common/svg_images_components/payment_success_standard_svg';
+import PaymentFailedSvg from 'components/common/svg_images_components/payment_failed_svg';
+import {getSubscriptionProduct} from 'mattermost-redux/selectors/entities/cloud';
+import {isCloudLicense} from 'utils/license_utils';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import {trackEvent} from 'actions/telemetry_actions';
+import useGetSubscription from 'components/common/hooks/useGetSubscription';
+
+type Props = {
+    callerCTA: string;
+}
+
+export default function DeleteWorkspaceModal(props: Props) {
+    const dispatch = useDispatch();
+    const openDowngradeModal = useOpenDowngradeModal();
+
+    // License/product checks.
+    const subscription = useGetSubscription();
+    const product = useSelector(getSubscriptionProduct);
+    const isStarter = product?.sku === CloudProducts.STARTER;
+    const isEnterprise = product?.sku === CloudProducts.ENTERPRISE;
+    const license = useSelector(getLicense);
+    const isNotCloud = !isCloudLicense(license);
+
+    // Starter product for downgrade purposes.
+    const starterProduct = useSelector((state: GlobalState) => {
+        return Object.values(state.entities.cloud.products || {}).find((product) => {
+            return product.sku === CloudProducts.STARTER;
+        });
+    });
+
+    // Get usage information in an attempt to defer customer from deleting.
+    const usage = useGetUsage();
+    const totalFileSize = fileSizeToString(usage.files.totalStorage);
+    const totalMessages = useSelector((state: GlobalState) => {
+        if (!state.entities.admin.analytics) {
+            return 0;
+        }
+        return state.entities.admin.analytics[StatTypes.TOTAL_POSTS];
+    });
+
+    // Creates a success modal for deletions, with button to redirect to mattermost.com.
+    const deleteSuccessModal = () => {
+        const handleButtonClick = () => {
+            window.open(MattermostLink, '_blank');
```

Comment:
I haven't stayed on top of the decision of whether deletes happen immediately or whether sessions get terminated immediately, but should we leave the current window open to the mattermost session in this case?

Reply:
I don't think there's been much discussion about this. The doc isn't too clear:

> Once all steps are completed, the workspace is “deleted”, or pending-deletion for 7 days (or the same number of days when a workspace is pending-deleted).

@johndavidlugtu Can we confirm if the workspace is meant to be "pending-deletion" for 7 days after deleting, or should the deletion occur immediately?

Reply:
From a small discussion with Nick, from the user's POV deletion is immediate, so the sessions should be terminated I think. I'll see if I can remove the exit button on the success result modal.
I would assume all user sessions would be terminated from the back-end side of things. Not sure if Allan's back-end PRs does this or if it happens automatically with the deletion request (I'd assume it doesn't since it's being sent to CWS if I recall correctly).